ui/src/app/route-reuse.service.ts

```diff
@@ -0,0 +1,52 @@
+import {Injectable} from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  DetachedRouteHandle,
+  RouteReuseStrategy,
+  Routes
+} from '@angular/router';
+
+// Configure reuse of particular routes so components do not have to re-load
+@Injectable()
+export class RouteReuse implements RouteReuseStrategy {
```

Comment:
Is this testable or does the test routing module not cover this stuff? (unit tests for this class by itself are not worth it)

Reply:
I tried testing it in the app.component.spec but unfortunately with no luck, it didn't seem to get triggered for some reason.